File: getrealweather/CheckWX.py
import requests_cache

class CheckWX(object):
    session = None
    apikey = None
    baseURL = 'https://api.checkwx.com/'

    def __init__(self, apikey):
        self.apikey = apikey
        self.session = requests_cache.CachedSession('checkwx_cache', backend='sqlite', expire_after=900)

    def getWeatherForLatLon(self, latitude, longitude):
        hdr = {'X-API-Key': self.apikey}
        requesturl = self.baseURL+f'metar/lat/{latitude}/lon/{longitude}/radius/100/decoded'
        weatherdata = self.session.get(requesturl, headers=hdr)
        return weatherdata

File: src/nodejs/weather/CheckWX.js
"use strict";

const got = require('got');

class CheckWX {

  constructor(apiKey) {
    if (!apiKey) {
      console.error("CheckWX API Key not defined; go get one on https://www.checkwxapi.com");
      process.exit(-1);
    }
    this._apiKey = apiKey
  }

  get apiKey() {
    return this._apiKey;
  }

  set apiKey(value) {
    this._apiKey = value;
  }

  getWeatherForLatLon(latitude, longitude) {
    return new Promise((resolve, reject) => {
      const requesturl = `${CheckWX.baseURL}metar/lat/${latitude}/lon/${longitude}/radius/100/decoded`;
      const options = {
        headers: {
          "Accept": "application/json",
          "X-API-Key": this._apiKey
        },
        timeout: {
          lookup: 1000,
          connect: 500,
          secureConnect: 500,
          socket: 10000,
          send: 10000,
          response: 10000
        }
      };

      (async () => {
        try {
          const response = await got(requesturl, options);
          let responseBodyJson = JSON.parse(response.body);
          resolve(responseBodyJson);
        } catch (error) {
          reject(error);
        }
      })();

    });
  }
}
CheckWX.baseURL = "https://api.checkwx.com/";

module.exports = CheckWX;